### PR TITLE
avm2: Use render_list instead of depth_list in most cases

### PR DESCRIPTION
**Pull request by Aaron1011**
The `render_list` for a container always contains all of the children
under both AVM1 and AVM2 - howver, the depth_list may not contain
some children under AVM2.

When we're not performing some AVM1-specific operation
(e.g. `getInstanceAtDepth`, or dumping out AVM1 variables),
we should be using the render list.

### CODE REVIEW

File: core/src/display_object/container.rs [hunk 1]
```diff
@@ -289,6 +289,10 @@ pub trait TDisplayObjectContainer<'gc>:
     }
 
     /// Iterates over the children of this display object in depth order.
+    /// Unless you are performing an AVM1-specific operation, you should
```
Comment (relrelb) [inline core/src/display_object/container.rs:292]: The only remaining usage of `iter_depth_list` is the Ctrl+Alt+V AVM1 `VariableDumper` in `player.rs`. I think it's a good opportunity to greatly simplify `iter_depth_list` (along with `DepthIter` and `lowest_depth`), maybe by iterating over an `i32` and call `child_by_depth` each time?

Comment (relrelb) [inline core/src/display_object/container.rs:292]: The only remaining usage of `iter_depth_list` is the Ctrl+Alt+V AVM1 `VariableDumper` in `player.rs`. I think it's a good opportunity to greatly simplify `iter_depth_list` (along with `DepthIter` and `lowest_depth`), maybe by iterating over an `i32` and call `child_by_depth` each time?

Comment (Aaron1011) [reply core/src/display_object/container.rs:292]: I considered that, but I was concerned that we might need it again in the future. If you think it's a good idea, then I'll go ahead and remove it.

Comment (relrelb) [reply core/src/display_object/container.rs:292]: I think `iter_depth_list` is not used not by chance - Flash probably doesn't have that functionality either. So I feel quite safe to remove it. Also, it's likely that `iter_children_by_depth` can fit as an alternative for the AVM1 `VariableDumper`.

Comment (Aaron1011) [reply core/src/display_object/container.rs]: I used `iter_render_list` and manually retrieved the depth - under AVM1, the depth and render lists should be same.

File: core/src/display_object/container.rs [hunk 2]
```diff
@@ -951,74 +937,3 @@ impl<'gc> DoubleEndedIterator for RenderIter<'gc> {
...
-            return None;
-        }
-
-        let mut old_depth = self.depth.unwrap();
-        let (new_depth, new_next) = if let Some(next) = self.src.child_by_depth(old_depth) {
-            (self.src.lowest_depth(old_depth), Some(next))
-        } else if let Some(new_depth) = self.src.lowest_depth(old_depth) {
-            old_depth = new_depth;
-            (
-                self.src.lowest_depth(new_depth),
```
Comment (relrelb) [inline core/src/display_object/container.rs:988]: `lowest_depth` is now unused and can be removed as well.

**Review by relrelb — APPROVED** [review]

**Comment by relrelb** [comment]
Thank you!

**Pull request merged**